src/composables/useRoutesMap.ts

```diff
@@ -0,0 +1,227 @@
+// libraries
+import { colors } from 'quasar';
+import { onMounted, provide, ref, unref } from 'vue';
+import { Map } from 'vue3-openlayers';
+import { LineString, MultiPoint, Point } from 'ol/geom';
+import { Style, Stroke, Icon } from 'ol/style';
+import { getLength } from 'ol/sphere';
+import { fromLonLat, useGeographic } from 'ol/proj';
+
+// composables
+import { i18n } from '../boot/i18n';
+
+// types
+import type { Ref } from 'vue';
+import type { Coordinate } from 'ol/coordinate';
+import type { Extent } from 'ol/extent';
+import type { Feature } from 'ol';
+import type { OverrideStyleFunction } from 'vue3-openlayers/dist/components/styles';
+import type { FeatureRoute } from '../components/types/Route';
+
+const { getPaletteColor } = colors;
+const primaryColor = getPaletteColor('primary');
+
+export const useRoutesMap = () => {
+  // provide ol options to child components
+  const isProduction = process.env.NODE_ENV === 'production';
+  const options = {
+    debug: !isProduction,
+  };
+  provide('ol-options', options);
+
+  // constants
+  const MAX_ZOOM_CENTERING_FACTOR = 15;
+  const DEFAULT_MAP_ZOOM = 13;
+
+  const mapRef = ref<InstanceType<typeof Map.OlMap> | null>(null);
+  const center = ref<Coordinate>(fromLonLat([14.4378, 50.0755]));
```

Comment:
Map center coordinate should be global variable.

Reply:
Moved to global config.